src/key.js

```diff
@@ -623,6 +623,86 @@ Key.prototype.revoke = function() {
 };
 
 /**
+ * Signs the public key
+ * @param  {Array<module:key~Key>} privateKey decrypted private keys for signing
+ * @return {module:key~Key} new public key with new certificate signature
+ */
+Key.prototype.sign = function(privateKeys) {
+  var primaryUser, primaryKey, dataToSign, signingKeyPacket, signaturePacket, user, key;
+  if (this.isPrivate()) {
+    throw new Error('Only public key can be signed');
+  }
+  primaryUser = this.getPrimaryUser();
+  if (!primaryUser) {
+    throw new Error('Could not find primary user');
+  }
+  primaryKey = this.primaryKey;
+  dataToSign = {};
+  dataToSign.userid = primaryUser.user.userId || primaryUser.user.userAttribute;
+  dataToSign.key = primaryKey;
```

Comment:
Correct, the thing you do on key signing parties is normally to sign User-Ids. There are also more exotic use cases for "direct signatures" on keys, but that should be out of scope here.
So I would propose to have a User.prototype.sign method (one private key as a parameter should be enough, I don't see the use case for an array). We could have helper methods like Key.prototype.signAllUser and Key.prototype.signPrimaryUser.
On the verification side we should go for a User.prototype.verifyOtherCertifications (or better name) and an array of keys as parameter. Here we have the same problem as with Message.prototype.verify: you don't know in advance which keys to hand over. Basically you have to look in Users.otherCertifications to determine the keyids, or you hand over your complete keyring. I once experimented with a method to supply the keys via a callback (https://github.com/mailvelope/openpgpjs/commit/4ffa64ae3e7507245ab9e2fadba3c1d25ccbc721), but have not gone further here.
As a return parameter for verifyOtherCertifications I'm leaning towards the same interface as used in Message.prototype.verify: true, false, null.